lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -455,52 +472,212 @@ class TextEditingElement {
     }));
   }
 
-  void setEditingState(EditingState editingState) {
-    _lastEditingState = editingState;
-    if (!isEnabled || !editingState.isValid) {
-      return;
+  @override
+  void updateElementPosition(_GeometricInfo geometricInfo) {
+    _geometricInfo = geometricInfo;
+    if (isEnabled && _canPosition) {
+      positionElement();
     }
+  }
 
-    _lastEditingState.applyToDomElement(domElement);
+  @override
+  void disableTextEditing() {
+    super.disableTextEditing();
+    _positionInputElementTimer?.cancel();
+    _positionInputElementTimer = null;
+  }
+}
 
-    // Re-focuses when setting editing state.
-    domElement.focus();
+/// Android behaviour for text editing.
+///
+/// inputmodeAttribute needs to be set for mobile devices. Due to this
+/// [initializeTextEditing] is different.
+///
+/// Keyboard acts differently than other devices. [addEventHandlers] handles
+/// this case as an extra.
+class AndroidTextEditingStrategy extends DefaultTextEditingStrategy {
+  @override
+  void initializeTextEditing(
+    InputConfiguration inputConfig, {
+    @required _OnChangeCallback onChange,
+    @required _OnActionCallback onAction,
+  }) {
+    super.initializeTextEditing(inputConfig,
+        onChange: onChange, onAction: onAction);
+    inputConfig.inputType.configureDomElement(domElement);
   }
 
-  void setGeometricInfo(_GeometricInfo geometricInfo) {
-    _geometricInfo = geometricInfo;
-    if (isEnabled) {
-      _positionElement();
+  @override
+  void addEventHandlers() {
+    super.addEventHandlers();
+    // Chrome on Android will hide the onscreen keyboard when you tap outside
+    // the text box. Instead, we want the framework to tell us to hide the
+    // keyboard via `TextInput.clearClient` or `TextInput.hide`.
+    //
+    // Safari on iOS does not hide the keyboard as a side-effect of tapping
+    // outside the editable box. Instead it provides an explicit "done" button,
+    // which is reported as "blur", so we must not reacquire focus when we see
+    // a "blur" event and let the keyboard disappear.
```

Comment:
Is this comment about iOS still relevant here?

Reply:
no I just carried around the comments in the first pass.

Now I fixed all the comments though. please have a look.